src/main/kotlin/io/github/gradlenexus/publishplugin/GetStagingProfile.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import org.gradle.api.GradleException
+import org.gradle.api.Incubating
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import javax.inject.Inject
+
+/**
+ * Gradle task to retrieve the staging profile id for a given [repository], either from the configured
+ * [NexusRepository.stagingProfileId] or via the remote [NexusRepository] HTTP API by using the [NexusClient].
+ */
+@Suppress("UnstableApiUsage")
+@Incubating
+open class GetStagingProfile @Inject constructor(
+    objects: ObjectFactory,
+    extension: NexusPublishExtension,
+    repository: NexusRepository
+) : AbstractNexusStagingRepositoryTask(objects, extension, repository) {
+
+    /**
+     * This property is [Optional] because it is only required if [NexusRepository.stagingProfileId] is not provided
+     */
+    @Input
+    @Optional
+    val packageGroup = objects.property<String>().apply {
+        set(extension.packageGroup)
+    }
+
+    @OutputFile
+    val stagingProfileIdOutput: RegularFileProperty = objects.fileProperty()
+
+    @TaskAction
+    fun getStagingProfile() {
+        val repository = repository.get()
+
+        val configuredStagingProfileId = repository.stagingProfileId.orNull
+        val stagingProfileId: String = if (configuredStagingProfileId == null) {
+            val client = NexusClient(
+                repository.nexusUrl.get(),
+                repository.username.orNull,
+                repository.password.orNull,
+                clientTimeout.orNull,
+                connectTimeout.orNull
+            )
```

Comment:
I guess this should be factored into an `internal val` since the client instantiation is duplicated in all the tasks.

Reply:
I believe, I proposed it in one of my earlier comments, but suggested to move it to a separate PR, after this one is merged to do not mix the things.

Reply:
Created #69.